### ISSUE TITLE
AP=0

### ISSUE THREAD

**Issue by 1757525671** [fundamentalvision/Deformable-DETR#91]
Ap of the pre-training model is normal, and ap of the finetune model is 0. Besides, loss decreases normally, and class_error also decreases, the reduction of learning rate will not solve it, very awful. I suspect something is wrong with the code.

**Comment by mei123456789** [comment]
@1757525671 Hello, are you using your own data set？

**Comment by 1757525671** [comment]
> @1757525671 Hello, are you using your own data set？

yes, My dataset has 2 categories, meanwhile, I set num_classes=2. I looked at the code all day yesterday and I just couldn't see what was wrong.

**Comment by mei123456789** [comment]
@1757525671 You can practice a few more rounds, it should improve

**Comment by mei123456789** [comment]
@1757525671 I am also having problems training my own data set and I am reporting an error that dimension 91 and 2 do not match, but my NUM_class has changed, can you tell me how to fix it? thank you

**Comment by 1757525671** [comment]
> @1757525671 You can practice a few more rounds, it should improve

Have you solved this problem?  I trained 34epoch, but AP =0.058............

**Comment by mei123456789** [comment]
> > @1757525671 You can practice a few more rounds, it should improve
> 
> Have you solved this problem? I trained 34epoch, but AP =0.058............

I can train directly. I can train for 50 rounds; But WHEN I used the pre-training weights, I had a mismatch of 91 and 2

**Comment by mei123456789** [comment]
@1757525671 I need your help with a question. What did you change with pre-training

**Comment by 1757525671** [comment]
> @1757525671 I am also having problems training my own data set and I am reporting an error that dimension 91 and 2 do not match, but my NUM_class has changed, can you tell me how to fix it? thank you

import torch

pretrained_weights = torch.load("./detr-r50-e632da11.pth")

num_class = 2
pretrained_weights["model"]["class_embed.weight"].resize_(num_class+1,256)
pretrained_weights["model"]["class_embed.bias"].resize_(num_class+1)

torch.save(pretrained_weights,'detr_r50_%d.pth'%num_class)


but the author says there is no need for a pre-training model to train its own data.https://github.com/fundamentalvision/Deformable-DETR/issues/21#issuecomment-752316544

**Comment by mei123456789** [comment]
> > @1757525671 I am also having problems training my own data set and I am reporting an error that dimension 91 and 2 do not match, but my NUM_class has changed, can you tell me how to fix it? thank you
> 
> import torch
> 
> pretrained_weights = torch.load("./detr-r50-e632da11.pth")
> 
> num_class = 2
> pretrained_weights["model"]["class_embed.weight"].resize_(num_class+1,256)
> pretrained_weights["model"]["class_embed.bias"].resize_(num_class+1)
> 
> torch.save(pretrained_weights,'detr_r50_%d.pth'%num_class)
> 
> but the author says there is no need for a pre-training model to train its own data.[#21 (comment)](https://github.com/fundamentalvision/Deformable-DETR/issues/21#issuecomment-752316544)

This is not the change of DETR, I changed to generate their own pre-training, but the problem occurred when running training

**Comment by mei123456789** [comment]
> > @1757525671 I am also having problems training my own data set and I am reporting an error that dimension 91 and 2 do not match, but my NUM_class has changed, can you tell me how to fix it? thank you
> 
> import torch
> 
> pretrained_weights = torch.load("./detr-r50-e632da11.pth")
> 
> num_class = 2
> pretrained_weights["model"]["class_embed.weight"].resize_(num_class+1,256)
> pretrained_weights["model"]["class_embed.bias"].resize_(num_class+1)
> 
> torch.save(pretrained_weights,'detr_r50_%d.pth'%num_class)
> 
> but the author says there is no need for a pre-training model to train its own data.[#21 (comment)](https://github.com/fundamentalvision/Deformable-DETR/issues/21#issuecomment-752316544)

This is  the change of DETR, I changed to generate their own pre-training, but the problem occurred when running training

**Comment by mei123456789** [comment]
@1757525671 Can I add your contact number for more details

**Comment by 1757525671** [comment]
> @1757525671 Can I add your contact number for more details

qq:1757525671

**Comment by ghost** [comment]
Hi @1757525671 , can you give the source for the pre-trained model?
I meed some problem with this.
Thank you in advance.

**Comment by MrYu001** [comment]
> > > @1757525671 I am also having problems training my own data set and I am reporting an error that dimension 91 and 2 do not match, but my NUM_class has changed, can you tell me how to fix it? thank you
> > 
> > 
> > import torch
> > pretrained_weights = torch.load("./detr-r50-e632da11.pth")
> > num_class = 2
> > pretrained_weights["model"]["class_embed.weight"].resize_(num_class+1,256)
> > pretrained_weights["model"]["class_embed.bias"].resize_(num_class+1)
> > torch.save(pretrained_weights,'detr_r50_%d.pth'%num_class)
> > but the author says there is no need for a pre-training model to train its own data.[#21 (comment)](https://github.com/fundamentalvision/Deformable-DETR/issues/21#issuecomment-752316544)
> 
> This is the change of DETR, I changed to generate their own pre-training, but the problem occurred when running training

Hi bro, when you load the pre-trained model weights, you just need to skip the final "class layer", or adjust the number of classes as your custom datasets, as 2 as you provided.
